task/task_smoke_test.go

```diff
@@ -19,19 +19,51 @@ import (
 	"terraform-provider-iterative/task/common"
 )
 
+type testTarget struct {
+	machine string
+	image   string
+	env     map[string]string
+	spot    common.Spot
+}
+
+var testTargets = map[string]testTarget{
+	// default test target spins up small instances and runs lightweight scripts on them.
+	"default": testTarget{
+		machine: "m",
+		image:   "ubuntu",
+		spot:    common.SpotDisabled,
+	},
+	"gpu": testTarget{
+		machine: "m+t4",
+		image:   "nvidia",
+		env: map[string]string{
+			"TEST_GPU": "yes",
+		},
+		spot: common.SpotEnabled,
+	},
```

Comment:
Cramming the GPU and spot tests under the `gpu` target migh be a bit too much.

I'd either:
1. Rename targets to e.g. “short” and “long” or “fast” and “slow” or “partial” and “full” or “consistent” and “flaky” ...
2. Create separate targets for both `gpu` and `spot` options and make them somehow composable.

Reply:
Ok, added, `quick`, `quick+spot`, `gpu` and `gpu+spot` as test targets.

Reply:
Thanks!